system_metrics_collector/src/topic_statistics_collector/topic_statistics_collector.hpp

```diff
@@ -35,12 +35,26 @@ class TopicStatisticsCollector : public system_metrics_collector::Collector
   TopicStatisticsCollector() = default;
   virtual ~TopicStatisticsCollector() = default;
 
-private:
   /**
    * Handle receiving a single message of type T.
+   *
    * @tparam T the ROS2 message type to collect
    */
   virtual void OnMessageReceived(const T & received_message) = 0;
+
+  /**
+   * Return the name to use for collected statistic
+   *
+   * @return a string of the name for this statistic
+   */
+  virtual const std::string GetStatisticName() const = 0;
```

Comment:
These duplicate the interface found in ::MetricsMessagePublisher. Instead of duplicating code, we should consider using the interface. If the publish method is too restrictive (I think it is), now is the time to remove it from the interface definition.

Reply:
Yeah it is very similar to the interfaces in `MetricsMessagePublisher`. Instead of removing the `PublishStatisticMessage()` method from there, would it make sense to separate the metric name and unit methods into a different interface? Then the `Collector` and `MetricsMessagePublisher` can both inherit it.

Reply:
That sounds good.